File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOneStringValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OneStringInValue.OneStringValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOneStringValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOneStringValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                String.class,
                OneStringValue.class.getDeclaredField("oneString").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                String.class,
                OneStringValue.class.getMethod("getOneString").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OneStringValue.class,
                OneStringValue.class.getMethod("setOneString", String.class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertEquals(
                "",
                new OneStringValue().getOneString());
    }
}

File: legacy/workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOneStringValue.scala
package com.dslplatform.ocd.values

import OneStringInValue.OneStringValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOneStringValue extends SpecExtensions {
  def `Test property field type: String`() =
    checkType(
      typeOf[String]
    , typeOf[OneStringValue].member("oneString": TermName).asMethod.returnType
    )

  def `Test property default value: ""`() {
    assert(OneStringValue().oneString === "")
  }
}
